src/client/src/apps/utils/openfin-utils.ts

```diff
@@ -31,25 +32,27 @@ async function restoreExistingApp(existingApp: Application): Promise<void> {
 export async function createOrBringToFrontOpenFinApplication({
   name,
   url,
+  uuid,
   windowOptions,
 }: ApplicationConfig): Promise<Application> {
   const existingApp = await getExistingOpenFinApplication(name)
   if (existingApp) {
     await restoreExistingApp(existingApp)
     return existingApp
   }
-  return createAndRunOpenFinApplication({ name, url, windowOptions })
+  return createAndRunOpenFinApplication({ name, url, uuid, windowOptions })
 }
 
 export async function createAndRunOpenFinApplication({
   name,
   url,
+  uuid,
   windowOptions,
 }: ApplicationConfig): Promise<Application> {
   const appOptions: fin.ApplicationOption = {
     name,
     url,
-    uuid: name,
+    uuid: uuid!,
```

Comment:
Thinking about this now @mattydono, you probably want to fallback to the original `name` arg. e.g `uuid: uuid || name,`

Reply:
ah yes that makes a lot more sense!